File: android/src/main/java/com/reactlibrary/RCTAlphaVideoManager.java
// RCTAlphaVideoManager.java
package com.reactlibrary;

import android.app.Activity;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

import java.util.Map;

public class RCTAlphaVideoManager extends SimpleViewManager<AlphaVideoView> {

    private static final String TAG = "RCTAlphaVideoManager";
    private AlphaVideoView videoView;

    // 事件名,这里写个enum方便循环
    public enum Events {
        //        EVENT_CODE_TYPES("codeTypes"),
        ON_DID_PLAY_FINISH("onDidPlayFinish");

        private final String mName;

        Events(final String name) {
            mName = name;
        }

        @Override
        public String toString() {
            return mName;
        }
    }

    public static final String REACT_CLASS = "RNAlphaVideo";

    /**
     * 设置别名
     *
     * @return
     */
    @Override
    public String getName() {
        return REACT_CLASS;
    }

    /**
     * 初始化入口
     *
     * @param context
     * @return
     */
    @NonNull
    @Override
    protected AlphaVideoView createViewInstance(@NonNull ThemedReactContext context) {
        Activity activity = context.getCurrentActivity();
        videoView = new AlphaVideoView(activity, context);
        Log.d(TAG, "createViewInstance: 初始化入口");
        return videoView;
    }

    /**
     * 注册事件
     *
     * @return
     */
    @Override
    @Nullable
    public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
        MapBuilder.Builder<String, Object> builder = MapBuilder.builder();
        for (Events event : Events.values()) {
            builder.put(event.toString(), MapBuilder.of("registrationName", event.toString()));
        }
        return builder.build();
    }

}

File: android/src/main/java/com/alphavideo/RCTAlphaVideoManager.kt
// RCTAlphaVideoManager.java
package com.alphavideo

import android.util.Log
import com.facebook.react.common.MapBuilder
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class RCTAlphaVideoManager : SimpleViewManager<AlphaVideoView>() {
    lateinit var context: ThemedReactContext

    // 事件名,这里写个enum方便循环
    enum class Events(private val mName: String) {
        //        EVENT_CODE_TYPES("codeTypes"),
        ON_DID_PLAY_FINISH("onDidPlayFinish");

        override fun toString(): String {
            return mName
        }
    }

    /**
     * 设置别名
     *
     * @return
     */
    override fun getName(): String {
        return REACT_CLASS
    }

    /**
     * 初始化入口
     *
     * @param context
     * @return
     */
    override fun createViewInstance(context: ThemedReactContext): AlphaVideoView {
        val activity = context.currentActivity
        this.context = context
        videoView = AlphaVideoView(activity!!, context)
        Log.d(TAG, "createViewInstance: 初始化入口")
        return videoView
    }

    @ReactProp(name = "source")
    fun setSource(view: AlphaVideoView, source: String) {
        if (source.startsWith("http")) {
            AlphaVideoParser.playVideoFromUrl(source, true) {
                videoView.getMxVideoView().setVideoFromSD(it)
            }
        } else {
            val resid = context.resources.getIdentifier(source, "raw", context.packageName)
            val afd = context.resources.openRawResourceFd(resid)
            videoView.getMxVideoView().setVideoFromFile(afd.fileDescriptor, afd.startOffset, afd.length)
            afd.close()
        }
    }

    @ReactProp(name = "loop", defaultBoolean = false)
    fun setLoop(view: AlphaVideoView, isLoops: Boolean) {
        view.getMxVideoView().setLooping(isLoops)
    }

    /**
     * 注册事件
     *
     * @return
     */
    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
        val builder = MapBuilder.builder<String, Any>()
        for (event in Events.values()) {
            builder.put(event.toString(), MapBuilder.of("registrationName", event.toString()))
        }
        return builder.build()
    }

    companion object {
        private const val TAG = "RCTAlphaVideoManager"
        const val REACT_CLASS = "RNAlphaVideo"

        lateinit var videoView: AlphaVideoView

    }
}